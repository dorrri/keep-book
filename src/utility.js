export const LIST_VIEW='list';
export const CHART_VIEW='chart';
export const TYPE_INCOME='income';
export const TYPE_OUTCOME='outcome';

export const padLeft=(n)=>{
	return n<10? '0'+n:n
};

export const range=(size, startAt=0)=>{
	const arr=[];
	for (let i=0;i<size;i++){
		arr[i]=startAt+i;
	}
	return arr
};

export const parseToYearAndMonth=(str)=>{
    const date=str?new Date(str):new Date();
    return {
    	year:date.getFullYear(),
		month:padLeft(date.getMonth()+1),
	}
};

export const isValidDate = (dateString) => {
	const regEx = /^\d{4}-\d{2}-\d{2}$/;
	if(!dateString.match(regEx)) return false;  // Invalid format
	const d = new Date(dateString);
	if(Number.isNaN(d.getTime())) return false; // Invalid date
	return d.toISOString().slice(0,10) === dateString;
};

export const flatternArr=(arr)=>{
	return arr.reduce((map,item)=>{
		map[item.id]=item;
		return map;
	},{})
};

export const calculateID = () => {
	// Math.random should be unique because of its seeding algorithm.
	// Convert it to base 36 (numbers + letters), and grab the first 9 characters
	// after the decimal.
	return '_' + Math.random().toString(36).substr(2, 9);
};

export const Colors = {
	blue: '#347eff',
	deepBlue: '#61dafb',
	green: '#28a745',
	red: '#dc3545',
	gray: '#555',
	lightGray: '#efefef',
	white: '#fff',
}