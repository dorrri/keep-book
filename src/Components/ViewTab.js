import React from 'react';
import Ionicon from 'react-ionicons';
import PropTypes from 'prop-types';
import {LIST_VIEW,CHART_VIEW} from '../utility'

const generateLinkClass=(current,view)=>{
   return (current===view)?'nav-link active':'nav-link'
};

const changeTab=(event)=>{

};
const ViewTab=({activeTab,onTabChange})=>
	(
	<ul className="nav nav-tabs nav-fill my-4">
		<li className="nav-item">
			<a
				href="#"
				className={generateLinkClass(activeTab,LIST_VIEW)}
				onClick={(event)=>{event.preventDefault(); onTabChange(LIST_VIEW)}}
			>
				<Ionicon
					className="rounded-circle mr-2"
					fontSize="30px"
					color={'#007bff'}
					icon='ios-paper'
				/>
				列表模式
			</a>
		</li>
		<li className="nav-item">
			<a
				href="#"
				className={generateLinkClass(activeTab,CHART_VIEW)}
				onClick={(event)=>{event.preventDefault(); onTabChange(CHART_VIEW)}}
			>
				<Ionicon
					className="rounded-circle mr-2"
					fontSize="30px"
					color={'#007bff'}
					icon='ios-pie'
				/>
				图表模式
			</a>
		</li>
	</ul>
);

ViewTab.propTypes={
	activeTab:PropTypes.string.isRequired,
	onTabChange:PropTypes.func.isRequired,
};

export default ViewTab;

