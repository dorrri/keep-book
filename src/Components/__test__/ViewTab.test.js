import React from 'react';
import {shallow} from 'enzyme';
import ViewTab from '../ViewTab'
import {LIST_VIEW,CHART_VIEW} from '../../utility'

const props={
    activeTab:LIST_VIEW,
	onTabChange:jest.fn(),
};

let wrapper;
describe('test ViewTab component',()=>{
	beforeEach(()=>{
		wrapper=shallow(<ViewTab {...props}/>);
	});
	it('should render the component to the snapshot', ()=> {
		expect(wrapper).toMatchSnapshot();
	});
	it('should active the correct tab', ()=> {
        expect(wrapper.find('.nav-item').first().find('a').props().className).toEqual('nav-link active');
	});
	// it('should trigger the correct function callbacks', ()=> {
	// 	const firstTab=wrapper.find('.nav-item').first().find('a').first();
	// 	firstTab.simulate('click');
	// 	expect(props.onTabChange).toHaveBeenCalled();
	// });
});