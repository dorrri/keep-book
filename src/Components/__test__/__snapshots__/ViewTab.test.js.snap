// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component should render the component to the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewTab
    activeTab="list"
    onTabChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="30px"
              icon="ios-paper"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            列表模式
          </a>
        </li>,
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="30px"
              icon="ios-pie"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            图表模式
          </a>
        </li>,
      ],
      "className": "nav nav-tabs nav-fill my-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="nav-link active"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="30px"
              icon="ios-paper"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            列表模式
          </a>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="30px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              />,
              "列表模式",
            ],
            "className": "nav-link active",
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "beat": false,
                "className": "rounded-circle mr-2",
                "color": "#007bff",
                "fontSize": "30px",
                "icon": "ios-paper",
                "rotate": false,
                "shake": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "列表模式",
          ],
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="nav-link"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="30px"
              icon="ios-pie"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            图表模式
          </a>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="30px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              />,
              "图表模式",
            ],
            "className": "nav-link",
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "beat": false,
                "className": "rounded-circle mr-2",
                "color": "#007bff",
                "fontSize": "30px",
                "icon": "ios-pie",
                "rotate": false,
                "shake": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "图表模式",
          ],
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item"
          >
            <a
              className="nav-link active"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="30px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              列表模式
            </a>
          </li>,
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="30px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              图表模式
            </a>
          </li>,
        ],
        "className": "nav nav-tabs nav-fill my-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="nav-link active"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="30px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              列表模式
            </a>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Ionicon
                  beat={false}
                  className="rounded-circle mr-2"
                  color="#007bff"
                  fontSize="30px"
                  icon="ios-paper"
                  rotate={false}
                  shake={false}
                  style={Object {}}
                />,
                "列表模式",
              ],
              "className": "nav-link active",
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "beat": false,
                  "className": "rounded-circle mr-2",
                  "color": "#007bff",
                  "fontSize": "30px",
                  "icon": "ios-paper",
                  "rotate": false,
                  "shake": false,
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "列表模式",
            ],
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="30px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              图表模式
            </a>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Ionicon
                  beat={false}
                  className="rounded-circle mr-2"
                  color="#007bff"
                  fontSize="30px"
                  icon="ios-pie"
                  rotate={false}
                  shake={false}
                  style={Object {}}
                />,
                "图表模式",
              ],
              "className": "nav-link",
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "beat": false,
                  "className": "rounded-circle mr-2",
                  "color": "#007bff",
                  "fontSize": "30px",
                  "icon": "ios-pie",
                  "rotate": false,
                  "shake": false,
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "图表模式",
            ],
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
